platforms/jvm/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/checkstyle/CheckstylePluginHtmlReportIntegrationTest.groovy

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.quality.checkstyle
+
+
+import org.gradle.integtests.fixtures.MultiVersionIntegrationSpec
+import org.gradle.integtests.fixtures.TargetCoverage
+import org.gradle.quality.integtest.fixtures.CheckstyleCoverage
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+
+@TargetCoverage({ CheckstyleCoverage.getSupportedVersionsByJdk() })
+class CheckstylePluginHtmlReportIntegrationTest extends MultiVersionIntegrationSpec {
```

Comment:
🎉 Thanks for adding this.  It still shocks me when I find something that's been around for years and years and has no test coverage.

💭 We might also want a test to validate that when the user sets a stylesheet, that stylesheet gets used instead of the default, but I don't think that's something we need to do now, unless you feel like you should.